server/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java

```diff
@@ -413,7 +413,7 @@ static class MaxFileSizeCheck implements BootstrapCheck {
         @Override
         public BootstrapCheckResult check(BootstrapContext context) {
             final long maxFileSize = getMaxFileSize();
-            if (maxFileSize != Long.MIN_VALUE && maxFileSize != ProcessLimits.UNLIMITED) {
+            if (maxFileSize != Long.MIN_VALUE && maxFileSize != -1 && maxFileSize != ProcessLimits.UNLIMITED) {
```

Comment:
Let's use the `ProcessLimits.UNKNOWN` constant?

Reply:
I would actually suggest splitting out the -1 change. It will require some more thought (and references to documentation), vs the simple and clear bugfix of calling the wrong method.

Reply:
`UNKNOWN` doesn't seem right. -1 means "no limit", not "unknown", according to [the docs](https://man.archlinux.org/man/limits.conf.5).

Reply:
Actually I don't think this fix is right. The value -1 should not be reaching this code per the javadocs on [`getRLimit`](https://github.com/elastic/elasticsearch/blob/53d8cc35694d2d14596d322878e780ed869610f1/libs/native/src/main/java/org/elasticsearch/nativeaccess/PosixNativeAccess.java#L54).